spyder/utils/programs.py

```diff
@@ -407,8 +423,11 @@ def is_module_installed(module_name, version=None, installed_version=None,
             get_modver = inspect.getsource(get_module_version)
             stable_ver = inspect.getsource(is_stable_version)
             ismod_inst = inspect.getsource(is_module_installed)
-            fd, script = tempfile.mkstemp(suffix='.py', dir=TEMPDIR)
-            with os.fdopen(fd, 'w') as f:
+
+            f = tempfile.NamedTemporaryFile('wt', suffix='.py', 
+                                            dir=get_temp_dir(), delete=False) 

```

Comment:
I really like this!! I didn't you could do that!

Reply:
It's a nice idiom... and in normal situations with delete=True (the default) you can forget about temporary files remaining in the filesystem